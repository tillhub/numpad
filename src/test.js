import React from 'react'
import NumPad from './'
import renderer from 'react-test-renderer'

const props = {
  handleChange: jest.fn()
}

describe('NumPad', () => {
  it('renders NumPad correctly', () => {
    const component = renderer.create(<NumPad {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders NumPad correctly in disabled mode', () => {
    const component = renderer.create(<NumPad {...props} disabled />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls function when a button is clicked', () => {
    const wrapper = mount(<NumPad {...props} />)
    const button7 = wrapper.find('span').at(2)

    expect(props.handleChange).toHaveBeenCalledTimes(0)

    button7.simulate('click')

    expect(toJson(wrapper)).toMatchSnapshot()
    expect(props.handleChange).toHaveBeenCalledTimes(1)
  })
})
