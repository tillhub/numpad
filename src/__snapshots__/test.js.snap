// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumPad calls function when a button is clicked 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <input
    className=""
    disabled={false}
    onChange={[Function]}
    value="7"
  />
  <div>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      <img
        src="backspace-icon.svg"
      />
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      C
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      7
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      8
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      9
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      4
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      5
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      6
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      1
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      2
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      3
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      0
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      .
    </span>
  </div>
</div>
`;

exports[`NumPad renders NumPad correctly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "width": "400px",
    }
  }
>
  <input
    className=""
    disabled={false}
    onChange={[Function]}
    value="0"
  />
  <div
    className={undefined}
  >
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      <img
        src="backspace-icon.svg"
      />
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      C
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      7
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      8
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      9
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      4
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      5
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      6
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      1
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      2
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      3
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      0
    </span>
    <span
      className=""
      disabled={false}
      onClick={[Function]}
    >
      .
    </span>
  </div>
</div>
`;

exports[`NumPad renders NumPad correctly in disabled mode 1`] = `
<div
  className={undefined}
  style={
    Object {
      "width": "400px",
    }
  }
>
  <input
    className=""
    disabled={true}
    onChange={[Function]}
    value="0"
  />
  <div
    className={undefined}
  >
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      <img
        src="backspace-icon.svg"
      />
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      C
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      7
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      8
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      9
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      4
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      5
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      6
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      1
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      2
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      3
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      0
    </span>
    <span
      className=""
      disabled={true}
      onClick={[Function]}
    >
      .
    </span>
  </div>
</div>
`;
